### DISCUSSION TITLE
How do I deal with Fixed Deposits

### DISCUSSION THREAD

**Discussion by anjalirai0** [ananthakumaran/paisa#102] [Q&A] [answered]
Let me explain, I've HDFC account and have a FD (10000Rs.) in the bank. Do I need to create a separate Debt account like (Assets:Debt:HDFC) where I show transactional entry of FD creation from Assets:Saving:HDFC and where will I receive interest, in Savings account or Debt Account?

**Comment by anjalirai0** [comment]
#!ledger
2022/11/11 Shriram Finance
	Assets:Debt:Shriram				200000.00 INR
 	Balance:Opening

	
I did this in main ledger for my old FD. Now this FD pays Rs.1201/- pm to my HDFC account, for which I make entries like:

2023/09/11 Shriram Finance
       Assets:Savings:HDFC                 1201.00 INR
       Income:Interest:Shriram


I feel something is wrong here. Can anyone please suggest corrections

**Reply by ananthakumaran** [reply, marked as answer]
```
2022/11/11 Opening balance
    Equity:OpeningBalances
    Assets:Checking:HDFC                     200,000 INR

2022/11/11 Open FD
    Assets:Checking:HDFC
    Assets:Debt:Shriram                      200,000 INR

2023/11/11 FD interest
    Income:Interest:Debt:Shriram
    Assets:Debt:Shriram                       20,100 INR

2023/11/11 Move interest to checking
    Assets:Debt:Shriram
    Assets:Checking:HDFC                      20,100 INR
```

> Do I need to create a separate Debt

Yes, usually it's better to create a new account to represent the FD. In my above example, I am starting with opening balance from Equity to Checking account. Equity should be used for these purposes. Now that you have money in your checking account, you can move it to FD, the date should match your FD opening date. Later if Shriram gives you interest, it should go to your FD account, from FD to Checking account. Even though it might go directly to your checking account in real life, this extra transaction is necessary for Paisa to calculate your returns from FD accurately. If you receive interest for any account, it should go to that account, you are free to move from that account to any other account.

![image](https://github.com/ananthakumaran/paisa/assets/149238/6525acaa-8c98-4f21-8156-97f481543274)

**Reply by anjalirai0** [reply]
Unrelated query, I'm trying to add Shares of Zomato to my ledger. But price update is not working. Am I missing something here.
I wrote following entry in ledger

2023/06/01 Investment
    Assets:Stocks:Zomato                     1 "ZOMATO" @    50.6764 INR
    Income:Salary:Globex

After this I went to configuration page and made following changes
<img width="1237" alt="Screenshot 2023-11-27 at 7 28 58 PM" src="https://github.com/ananthakumaran/paisa/assets/152173377/f087ca3e-99f4-4a6a-8874-ea8e417ec3ef">


But price is still not updating

**Reply by ananthakumaran** [reply]
Just tried it, and it should work

![image](https://github.com/ananthakumaran/paisa/assets/149238/13d9debd-f32b-431b-ae5e-1c769696ae44)


Make sure you have updated the price via the menu at the top right hand corner, Prices are not updated automatically and has to be done manually via the menu.

**Reply by anjalirai0** [reply]
Yes it finally worked.

**Comment by anjalirai0** [comment]
Not able to fix this issue. Zomato ticker symbol is stuck at AAPL.
i've already tried deleting backup files, restarting app, syncing journal, deleting and re-adding zomato in commodities. 

Need help with this

<img width="1440" alt="Screenshot 2023-11-27 at 10 54 22 PM" src="https://github.com/ananthakumaran/paisa/assets/152173377/fd0954a5-9cfc-4664-8387-0b38da430ea9">


<img width="955" alt="Screenshot 2023-11-27 at 10 55 51 PM" src="https://github.com/ananthakumaran/paisa/assets/152173377/7a9bfbc0-4e7b-4fb7-8fa4-65ad121cd600">


> Just tried it, and it should work
> 
> ![image](https://private-user-images.githubusercontent.com/149238/285900336-13d9debd-f32b-431b-ae5e-1c769696ae44.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDExMDYzMjQsIm5iZiI6MTcwMTEwNjAyNCwicGF0aCI6Ii8xNDkyMzgvMjg1OTAwMzM2LTEzZDlkZWJkLWYzMmItNDMxYi1hZTVlLTFjNzY5Njk2YWU0NC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMTI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTEyN1QxNzI3MDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03ZWJjMDJmMGQ0YTY2Yjc3ODJmZjE5ZGRiMWQ4YmM2NTViMDkzYTk3MjczNjRhOGE5Y2MyNDVlOThmMDJjZTM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.a6LddQkeVTvg85-MJD4ulDR94NzJlgbKp1fSLMRd3_I)
> 
> Make sure you have updated the price via the menu at the top right hand corner, Prices are not updated automatically and has to be done manually via the menu.

**Reply by ananthakumaran** [reply]
Looks like you found a bug, I will add an option to clear the price history in the next release. If you know how to use the command line, you can use the following steps to fix the problem

1) close the app
2) go to the folder where all the paisa-related files are stored, `cd ~/Documents/paisa` from the terminal should take you there
3) `sqlite3 paisa.db "delete from prices";` run this command from terminal

If not, you can try to use a different commodity name for now.

**Reply by ananthakumaran** [reply]
Another option is to just delete the `paisa.db` file in the Documents/paisa folder. Reopen the app and sync journal and update price after that, it should fix the issue

**Reply by anjalirai0** [reply]
I created new Db path and it worked. Thanks

I'm not sure what led to this issue

**Reply by ananthakumaran** [reply]
There is an issue with how commodity price update is handled, it's not clearing the old values, that cause the problem. I will fix it.
